app/code/core/Mage/Adminhtml/Helper/TwoFactorAuthentication.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * Maho
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2024 Maho (https://mahocommerce.com)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ */
+class Mage_Adminhtml_Helper_TwoFactorAuthentication extends Mage_Core_Helper_Abstract
+{
+    public function getSecret(): string
+    {
+        return \OTPHP\TOTP::create()->getSecret();
+    }
+
+    public function getQRCodeUrl(#[\SensitiveParameter] string $username, #[\SensitiveParameter] string $secret): string
+    {
+        $storeName = Mage::getStoreConfig('general/store_information/name');
+        $otp = \OTPHP\TOTP::create($secret);
+        $otp->setLabel($username . '@' . $storeName);
+        $otp->setIssuer('Maho Admin');
+
+        return $otp->getQrCodeUri(
+            'https://api.qrserver.com/v1/create-qr-code/?color=000000&bgcolor=FFFFFF&data=[DATA]&qzone=2&margin=0&size=300x300&ecc=M',
```

Comment:
I'm using this online service to generate the QR because I wouldn't want to deal with images to save and then delete etc... is that ok?

Reply:
Just mentioned that point in a comment. I think the SVG route could allow us to not have to save any files.